Add library system name & service point ID fields to Library

humble-ahitofel opened the pull request:
Add two basic fields without any smarter validation (library systems can have very specific id formats).
Note: API v4 has a separate merge request for this later.

ajsalminen commented on templates/entity/Library/edit.html.twig:
```diff
@@ -44,6 +44,15 @@
+            {{ form_row(form.library_system_id)}}
+          </div>
+        </div>
+        
```
```suggestion
```

ajsalminen commented on src/Module/ApiCache/Serializer/Normalizer/LibraryNormalizer.php:
```diff
@@ -78,13 +78,24 @@ public function normalize($object, $format = null, array $context = [])
             $values['customData'] = $this->extractCustomData($object->getCustomData());
         }
 
+        
+
```
Just a couple of unnecessary whitespace lines.

```suggestion
```

ajsalminen replied:
```suggestion
```

ajsalminen commented on data/database.alter.sql:
```diff
@@ -2176,3 +2176,10 @@ ALTER TABLE schedules DROP COLUMN staff;
+ALTER TABLE organisations ADD library_system_name varchar(255);
+COMMENT ON COLUMN organisations.library_system_name IS 'Machine name of the system in use (sierra, axiell, koha)';
+
+ALTER TABLE organisations ADD library_system_id varchar(255);
```
After our discussion about what this represents my thought here is that maybe it would be better to call this field "library_system_servicepoint_id". Would that be an appropriate name and are there any issues changing it? The benefit would be that it becomes clear at first look that it doesn't mean a library system (instance or system type) specific identifier but something that identifies some more specific part of the library operations like one service point.

The comment on the column could maybe be a little more specific about the purpose too.

humble-ahitofel replied:
Should renaming `library_system_id` to `library_system_servicepoint_id` reflect to everything else, like for example the kirkanta api (query param names, json payload, etc)?

ajsalminen replied:
Yeah I think that would be ideal if it would since it cuts through all those as the same thing as far as I understand, does doing that cause any additional issues?

humble-ahitofel replied:
No, not really. I will implement the changes now.

humble-ahitofel replied:
`library_system_id` has been now renamed. I revised a little the sql commenting.

ajsalminen commented on data/database.alter.sql:
```diff
@@ -2176,3 +2176,10 @@ ALTER TABLE schedules DROP COLUMN staff;
+
+
+ALTER TABLE organisations ADD library_system_name varchar(255);
+COMMENT ON COLUMN organisations.library_system_name IS 'Machine name of the library system that is used currently in the orgainsation (sierra, axiell, koha)';
```
```suggestion
COMMENT ON COLUMN organisations.library_system_name IS 'Machine name of the library system that is used currently in the organisation (sierra, axiell, koha)';
```

ajsalminen commented on data/database.alter.sql:
```diff
@@ -2176,3 +2176,10 @@ ALTER TABLE schedules DROP COLUMN staff;
+COMMENT ON COLUMN organisations.library_system_name IS 'Machine name of the library system that is used currently in the orgainsation (sierra, axiell, koha)';
+
+ALTER TABLE organisations ADD library_system_servicepoint_id varchar(255);
+COMMENT ON COLUMN organisations.library_system_servicepoint_id IS 'ID given by the library system to the organsation (Sierra, Axiell, Koha, etc.)';
```
```suggestion
COMMENT ON COLUMN organisations.library_system_servicepoint_id IS 'ID given by the library system to the organisation (Sierra, Axiell, Koha, etc.)';
```

humble-ahitofel committed:
Add library system info fields to Library (#36)

These fields record the library system and a system-specific identifier
for a library or a service point. This is used for combining data from
library directory and other APIs for statistical purposes. At first this
will be done with Axiell's API.

Note: The required filtering functionality for Library Directory API
is implemented in API v4. (libraries-fi/kirkanta-api#10)

ajsalminen commented:
I made some minor changes to documentation.md but didn't generate the html version Should be good to go after that, can you update it?
